include/atomics/receiver_cadmium.hpp

```diff
@@ -0,0 +1,114 @@
+/**
+* Cristina Ruiz Martin
+* ARSLab - Carleton University
+*
+*/
+
+#ifndef BOOST_SIMULATION_PDEVS_RECEIVER_HPP
+#define BOOST_SIMULATION_PDEVS_RECEIVER_HPP
+
+#include <cadmium/modeling/ports.hpp>
+#include <cadmium/modeling/message_bag.hpp>
+#include <limits>
+#include <math.h> 
+#include <assert.h>
+#include <memory>
+#include <iomanip>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <chrono>
+#include <algorithm>
+#include <limits>
+#include <random>
+
+//updated relative path --Syed Omar
+#include "../data_structures/message.hpp"
+
+using namespace cadmium;
+using namespace std;
+
+//Port definition
+// changed Receiver_defs to receiver_defs Syed Omar
+struct receiver_defs{
+    struct out : public out_port<message_t> {};
+    struct in : public in_port<message_t> {};
+};
+   
+template<typename TIME>
+class Receiver{
+    using defs = receiver_defs; // putting definitions in context
+    public:
+    //Parameters to be overwriten when instantiating the atomic model
+    TIME PREPARATION_TIME;
+    //default constructor
+    Receiver() noexcept{
+        PREPARATION_TIME = TIME("00:00:10");
+        state.ack_num = 0;
+        state.sending = false;
+    }
+            
+    // state definition
+    struct state_type{
+        int ack_num;
+        bool sending;
+    }; 
+    state_type state;
+    // ports definition
+    using input_ports = std::tuple<typename defs::in>;
+    using output_ports = std::tuple<typename defs::out>;
+
+    // internal transition
+    void internal_transition(){
+        state.sending = false; 
+    }
+
+    // external transition
+    void external_transition(TIME e, 
+                             typename make_message_bags<input_ports>::type mbs){ 

```

Comment:
Line is 82 in length so to solve that you can bring TIME e one line down or remove some spaces